### PR TITLE
fix(buildkite): Handle more use-cases in config parser

### LINKED ISSUE #3956: Buildkite plugins not updating
**What Renovate type are you using?**
Renovate Pro?

**Describe the bug**

I added in the preset for buildkite, but i receive no PRs for updates to the plugins.

From looking at renovate logs, it may be due to the fact I have an alias for my shared plugins.

**To Reproduce**

```yaml

plugins: &plugins #alias for shared plugins
  seek-oss/aws-sm#v0.0.3:
    env:
      NPM_TOKEN: "arn:aws:secretsmanager:ap-southeast-2:xxx:secret:npm/xxx"
  seek-oss/private-npm#v1.1.1:
    env: "NPM_TOKEN"
  docker#v3.0.1:
    volumes:
      - /app/node_modules
    workdir: "/app"  
  seek-oss/docker-ecr-cache#v1.1.4:
    dockerfile: Dockerfile-builder
    cache-on:
      - yarn.lock   
```
**Expected behavior**

Get PRs for new versions of the plugins

### PR DESCRIPTION
**Pull request by zharinov**
Closes #3956

Sample PRs: https://github.com/zharinov/buildkite-test/pulls

### CODE REVIEW

File: lib/manager/buildkite/extract.js [hunk 1]
```diff
@@ -9,17 +9,24 @@ function extractPackageFile(content) {
   const deps = [];
   try {
     const lines = content.split('\n');
-    for (let lineNumber = 0; lineNumber < lines.length; lineNumber += 1) {
-      const line = lines[lineNumber];
-      const plugins = line.match(/^\s*-?\s*plugins:\s*$/);
+    let pluginsSection = null;
+    let pluginsIndent = '';
+    for (let lineNumber = 1; lineNumber <= lines.length; lineNumber += 1) {
```
Comment (ViceIce) [inline lib/manager/buildkite/extract.js]: Why we can't start at zero?

Comment (zharinov) [reply lib/manager/buildkite/extract.js]: We log these numbers and my point was to make it consistent with code listings in editors/github/etc

**Review by rarkins — CHANGES_REQUESTED** [review]
Needs merge from master and de-conflicting

**Review by viceice — APPROVED** [review]

**Comment by rarkins** [comment]
@zharinov needs coverage fixed

**Pull request merged**